### ISSUE TITLE
GLPI from v9.1.3 to v9.2.1 - Truncated (ticket) URL link on webcal export task + last tasks not synced to CalDAV cal

### ISSUE THREAD

**Issue by Krisstof** [glpi-project/glpi#2133]
Version GLPI 9.1.3

2 problems
1/
Ticket URL in webcal export task is truncated 
found URL : /front/ticket.form.php?id=13800
It should be : https://server.domain/front/ticket.form.php?id=13800

2/
Another problem on webcal export : all tasks synced correctly on CalDAV until last 4 days. Since the last 4 days no more task synched on my calendar (CalDAV) and of course I have many tasks on GLPI planning. Same problem on others GLPI users calendars. Notes are correctly synced. Any idea ? 

<img width="430" alt="capture d ecran 2017-04-29 a 09 32 01" src="https://cloud.githubusercontent.com/assets/3646886/25553856/3da8aef0-2cbf-11e7-8a4f-9677cfe44738.png">

**Comment by Krisstof** [comment]
sql-errors.log sample 

2017-05-03 14:24:04 [153@domain.tld]
  *** MySQL query error:
  SQL: SELECT `glpi_projecttasks`.*
                FROM `glpi_projecttaskteams`
                INNER JOIN `glpi_projecttasks`
                  ON (`glpi_projecttasks`.`id` = `glpi_projecttaskteams`.`projecttasks_id`)
                WHERE `glpi_projecttaskteams`.`itemtype` = \'User\'
                       AND `glpi_projecttaskteams`.`items_id` = \'153\'
                       AND
                      \'2017-05-03 00:00\' < `glpi_projecttasks`.`plan_end_date`
                      AND \'2017-05-03 24:00\' > `glpi_projecttasks`.`plan_start_date`
                ORDER BY `glpi_projecttasks`.`plan_start_date`
  Error: Illegal mix of collations (utf8_general_ci,COERCIBLE) and (latin1_swedish_ci,NUMERIC) for operation '>'
  Backtrace :
  inc/projecttask.class.php:1311
  inc/planning.class.php:393                         ProjectTask::populatePlanning()
  front/planning.php:65                              Planning::checkAvailability()
2017-05-03 14:24:04 [153@domain.tld]
  *** MySQL query error:
  SQL: SELECT `glpi_plugin_resources_tasks`.*,
                        `glpi_plugin_resources_taskplannings`.`begin`,
                        `glpi_plugin_resources_taskplannings`.`end`,
                        `glpi_plugin_resources_resources`.`name` as resource,
                        `glpi_plugin_resources_tasktypes`.`name` as type
                FROM `glpi_plugin_resources_tasks`
                LEFT JOIN `glpi_plugin_resources_taskplannings` ON (`glpi_plugin_resources_taskplannings`.`plugin_resources_tasks_id` = `glpi_plugin_resources_tasks$
                LEFT JOIN `glpi_plugin_resources_resources`
                ON (`glpi_plugin_resources_resources`.`id` = `glpi_plugin_resources_tasks`.`plugin_resources_resources_id`)
                LEFT JOIN `glpi_plugin_resources_tasktypes`
                ON (`glpi_plugin_resources_tasktypes`.`id` = `glpi_plugin_resources_tasks`.`plugin_resources_tasktypes_id`)
                WHERE `glpi_plugin_resources_tasks`.`users_id` = \'153\'
                     AND
                      \'2017-05-03 00:00\' < `end` AND \'2017-05-03 24:00\' > `begin` AND `glpi_plugin_resources_tasks`.`is_finished` != 1
                ORDER BY `begin`
  Error: Illegal mix of collations (utf8_general_ci,COERCIBLE) and (latin1_swedish_ci,NUMERIC) for operation '>'
  Backtrace :
  plugins/resources/inc/taskplanning.class.php:326
  inc/planning.class.php:393                         PluginResourcesTaskPlanning::populatePlanning()
  front/planning.php:65                              Planning::checkAvailability()

**Comment by nqb** [comment]
In 9.1.4, first problem always present. In ICS file, we can see that : 
`URL:/glpi/front/ticket.form.php?id=18000`

**Comment by dariopsmw** [comment]
This also happens on 9.2. I have tested the query and this date '2017-05-03 24:00' is the problematic part of the sentence. "24:00" confuses mysql engine, which is invalid, should be '2017-05-04 00:00'

**Comment by Krisstof** [comment]
I'm glad to know that I'm not the only one person who use WebCal functionality (on Apple CalDAV for my use).
7 month later, not declared as bug and not schedule for fixe it.
Still 2 bugs : 
- Truncated URL in ICS
- About 2 days to sync tasks on CalDAV client.

**Comment by dariopsmw** [comment]
I'll probably just hack a replace ('24:00', '23:59') but i don't understand where yet. I couldn't follow the code, i'm not a developer.

**Comment by nqb** [comment]
+1 
@Krisstof : could you edit subject issue (9.1.3 => 9.2) ?

**Comment by nqb** [comment]
@orthagh : issue still presents in 9.2.1, could you provide a patch ?

**Comment by orthagh** [comment]
issue will be solved in 9.2.2, pr incoming

> 7 month later, not declared as bug and not schedule for fixe it.
we are a small team and we do our maximum. Please be patient.

**Comment by nqb** [comment]
@orthagh : patch applied.

I have full URL but they called `ticket.php` in place of `ticket.form.php`. Please change this behavior.

**Comment by trasher** [comment]
@nqb it should have been fixed. Could you please test again? Thank you.

**Comment by nqb** [comment]
yes @trasher.

**Comment by nqb** [comment]
@orthagh, @trasher : new patch applied.

1/ : all is right. `.form.php` added on each URL lines in ICS file.

Just one comment, I notice that `SUMMARY` lines are different between elements. For tickets, you have `Ticket #ID Title_of_ticket`. For problems or changes, you have `Title_of_change_or_problem`

2/ : not sure how to reproduce issue reported by @Krisstof. 

I try this : 
* Configure in Thunderbird a new calendar with this settings : 
  * On network
  * Format : ICS
  * Location : Link of "Export - Ical" button on my GLPI planning
=> I've got events until March 2017
* Edit a ticket task scheduled in September 2017 to change its description
* Sync my calendar in Thunderbird
=> I've got updated task on my Thunderbird (new description)
